be/src/storage/async_delta_writer_executor.h

```diff
@@ -7,28 +7,36 @@ DIAGNOSTIC_IGNORE("-Wclass-memaccess")
 DIAGNOSTIC_POP
 
 #include "common/config.h"
+#include "util/monotime.h"
 #include "util/threadpool.h"
 
 namespace starrocks {
+const int64_t RETRY_INTERVAL_MS = 50;
 
 // Used to run bthread::ExecutionQueue task in pthread instead of bthread.
 // Reference: https://github.com/apache/incubator-brpc/blob/master/docs/cn/execution_queue.md
 class AsyncDeltaWriterExecutor : public bthread::Executor {
 public:
-    Status init() {
+    Status init(int max_queue_size = 40960) {
         if (_thread_pool != nullptr) {
             return Status::InternalError("already initialized");
         }
         return ThreadPoolBuilder("delta_writer")
                 .set_min_threads(config::number_tablet_writer_threads / 2)
                 .set_max_threads(std::max<int>(1, config::number_tablet_writer_threads))
-                .set_max_queue_size(40960)
+                .set_max_queue_size(max_queue_size)
                 .set_idle_timeout(MonoDelta::FromMilliseconds(5 * 60 * 1000))
                 .build(&_thread_pool);
     }
 
     int submit(void* (*fn)(void*), void* args) override {
-        auto st = _thread_pool->submit_func([=]() { fn(args); });
+        Status st;
+        while (true) {
+            st = _thread_pool->submit_func([=]() { fn(args); });
+            if (!st.is_service_unavailable()) break;
+            LOG(INFO) << "async_delta_writer is busy, retry after ms: " << RETRY_INTERVAL_MS;
+            SleepFor(MonoDelta::FromMilliseconds(RETRY_INTERVAL_MS));
```

Comment:
According to my knowledge, this method may be involked in a bthread, calling `SleepFor` in bthread is bad. Maybe you can return EAGAIN on service unavailable and let the caller do the retry.

Reply:
`ExecutionQueueBase` calls `bthread::Executor::submit` and `PLOG(FATAL)` if  get any error.
https://github.com/apache/incubator-brpc/blob/1ad1cc57ecc50d4dd631e32781459830f75ca23a/src/bthread/execution_queue.cpp#L120

Reply:
If so, use [bthread_usleep](https://github.com/apache/incubator-brpc/blob/37a21f44f97ffe8f17bd542c2de4f974ddd16bbe/src/bthread/bthread.h#L156)

Reply:
I agree and have updated the code.

Reply:
If the brpc can not succeed, will it be an endless loop?
Does it seem like a BRPC bug?

Reply:
The problem is `_thread_pool` may return an error status on busy, it has nothing to do with brpc.

Reply:
> `ExecutionQueueBase` calls `bthread::Executor::submit` and `PLOG(FATAL)` if get any error. https://github.com/apache/incubator-brpc/blob/1ad1cc57ecc50d4dd631e32781459830f75ca23a/src/bthread/execution_queue.cpp#L120

The direct reason is that the user-defined `bthread::Executor::submit` cannot return non-zero. I think it's not good but seems no better solution.

Reply:
Got it. you can try to connect the brpc community ask if it's to change the log level from FATAL to ERROR.